openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -0,0 +1,446 @@
+
+    Args:
+        user_for_testing: User object
+        get_request_for_application_hub_view: HttpRequest for application hub 'get'
+
+    Returns:
+        Boolean
+
+    """
+    user = user_for_testing
+    ApplicationHub.objects.filter(user=user).update(is_prerequisite_courses_passed=True,
+                                                    is_written_application_completed=True,
+                                                    is_application_submitted=True)
+
+    response = ApplicationHubView.as_view()(get_request_for_application_hub_view)
+    assert response.get('Location') == reverse('application_success')
+    assert response.status_code == 302
+
+
+@pytest.mark.django_db
+def test_post_logged_in_user_without_required_objectives_completed_for_application_hub_view(user_for_testing,
+                                                                                            logged_in_client):
+    """
+    Test the case where an authenticated user hits the url without having completed the required objectives i.e
+    the pre_req courses and the written application.
+
+    Args:
+        user_for_testing: User object
+        logged_in_client: Client obj with logged in user
+
+    Returns:
+        Boolean
+
+    """
+    user = user_for_testing
+
+    ApplicationHub.objects.filter(user=user).update(is_prerequisite_courses_passed=False,
+                                                    is_written_application_completed=False)
+
+    response = logged_in_client.post(reverse('application_hub'))
+    assert response.status_code == 400
+
+
+@mock.patch('openedx.adg.lms.applications.views.send_mandrill_email')
+@mock.patch('openedx.adg.lms.applications.views.render')
+def test_post_logged_in_user_with_required_objectives_completed_to_application_hub_view(mock_render,
+                                                                                        mock_send_mandrill_mail,
+                                                                                        user_for_testing,
+                                                                                        logged_in_client):
+    """
+    Test the case where an authenticated user, with all the required objectives completed, hits the url.
+
+    Args:
+        mock_render: Mock object for render function
+        mock_send_mandrill_mail: Mock object for send_mandrill_mail function
+        user_for_testing: User object
+        logged_in_client: Client obj with logged in user
+
+    Returns:
+        Boolean
+
+    """
+    ApplicationHub.objects.filter(user=user_for_testing).update(is_prerequisite_courses_passed=True,
+                                                                is_written_application_completed=True)
+
+    response = logged_in_client.post(reverse('application_hub'))
+    user_application_hub = ApplicationHub.objects.get(user=user_for_testing)
+    assert user_application_hub.is_application_submitted is True
```

Comment:
is True is redundant as **assert user_application_hub.is_application_submitted** is enough.